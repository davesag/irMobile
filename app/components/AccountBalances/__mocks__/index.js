import React from 'react'
import { Text } from 'react-native'

export default jest.fn(props => <Text>Account Balances</Text>)
