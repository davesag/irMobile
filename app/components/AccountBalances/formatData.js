const LONGEST = {
  balance: undefined,
  fiatPrice: undefined,
  fiatValue: undefined,
  total: undefined
}

const reset = () => {
  const base = { lhs: 0, rhs: 0 }
  Object.keys(LONGEST).forEach(key => {
    LONGEST[key] = { ...base }
  })
}

const round = (number, decimals = 0) => {
  if (decimals === 0) return number
  const power = Math.pow(10, decimals)
  return Math.round(number * power + Number.EPSILON) / power
}

const toString = (value, decimals) => {
  const localeOptions = { type: 'decimal', maximumFractionDigits: 8 }
  if (decimals) localeOptions.minimumFractionDigits = 2

  return round(value, decimals).toLocaleString('en-AU', localeOptions)
}

const checkLongest = row => (field, decimals) => {
  const value = row[field]
  const string = toString(value, decimals)
  const [lhs, rhs] = string.split('.')
  if (lhs.length > LONGEST[field].lhs) LONGEST[field].lhs = lhs.length
  if (rhs !== undefined && rhs.length > LONGEST[field].rhs)
    LONGEST[field].rhs = rhs.length
  return string
}

const pad = field => string => {
  const [lhs, rhs] = string.split('.')
  const padLeft = lhs.padStart(LONGEST[field].lhs, ' ')
  const padRight = (rhs || '').padEnd(LONGEST[field].rhs, ' ')
  return rhs ? `${padLeft}.${padRight}` : `${padLeft} ${padRight}`
}

const formatData = data => {
  reset()
  const rows = data.map(row => [
    row.currency,
    checkLongest(row)('balance'),
    checkLongest(row)('fiatPrice', 3),
    checkLongest(row)('fiatValue', 2)
  ])

  const fValue = data.reduce((acc, { fiatValue }) => {
    acc = acc + fiatValue
    return acc
  }, 0)
  const total = checkLongest({ fiatValue: fValue })('fiatValue', 2)

  // { currency: 'Eth', balance: 21.62806365, fiatPrice: 464.04, fiatValue: 10036.286656146001 }

  const body = rows.reduce((acc, elem) => {
    const [currency, balance, fiatPrice, fiatValue] = elem
    acc.push([
      currency,
      pad('balance')(balance),
      pad('fiatPrice')(fiatPrice),
      pad('fiatValue')(fiatValue)
    ])
    return acc
  }, [])
  const footer = ['Total', '', '', total]
  return { body, footer }
}

export default formatData
