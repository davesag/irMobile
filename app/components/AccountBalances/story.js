import React from 'react'
import { storiesOf } from '@storybook/react-native'
import { SafeAreaView } from 'react-native'

import AccountBalances from '.'

const balances = [
  { currency: 'Xbt', balance: 3, fiatPrice: 16839.25, fiatValue: 50517.75 },
  {
    currency: 'Eth',
    balance: 21.62806365,
    fiatPrice: 464.04,
    fiatValue: 10036.286656146001
  }
]

storiesOf('AccountBalances', module).add('simple', () => (
  <SafeAreaView>
    <AccountBalances balances={balances} />
  </SafeAreaView>
))
