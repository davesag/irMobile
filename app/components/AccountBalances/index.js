import React, { Component } from 'react'
import PropTypes from 'prop-types'
import Table from '../Table'

import styles from './styles'
import formatData from './formatData'

const header = ['Currency', 'Balance', 'AUD Price', 'AUD Value']

class AccountBalances extends Component {
  static propTypes = {
    details: PropTypes.array.isRequired
  }

  // because we will update these fields later via sockets
  constructor(props) {
    super(props)

    this.state = {
      details: props.details
    }
  }

  render() {
    const { details } = this.state
    const { body, footer } = formatData(details)

    return (
      <Table
        tableStyle={styles.container}
        header={header}
        body={body}
        footer={footer}
      />
    )
  }
}

export default AccountBalances
