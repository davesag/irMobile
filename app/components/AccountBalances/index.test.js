import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'

import AccountBalances from '.'

const balances = [
  { currency: 'Xbt', balance: 3, fiatPrice: 16839.25, fiatValue: 50517.75 },
  {
    currency: 'Eth',
    balance: 21.62806365,
    fiatPrice: 464.04,
    fiatValue: 10036.286656146001
  }
]

let tree

jest.mock('../Table')

describe('rendering', () => {
  describe('defaults', () => {
    beforeAll(() => {
      tree = renderer.create(<AccountBalances balances={balances} />)
    })

    it('rendered correctly ', () => {
      expect(tree).toMatchSnapshot()
    })
  })
})
