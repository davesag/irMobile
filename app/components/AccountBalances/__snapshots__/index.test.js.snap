// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering defaults rendered correctly  1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "flex-start",
      },
      Object {
        "paddingTop": 10,
      },
    ]
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-evenly",
        "paddingTop": 2,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "alignSelf": "center",
          "borderBottomWidth": 0.5,
          "flexWrap": "nowrap",
          "height": 25,
          "justifyContent": "center",
          "overflow": "hidden",
          "paddingBottom": 3,
          "paddingLeft": 5,
          "paddingRight": 5,
          "width": "25%",
        }
      }
    >
      <Text
        style={
          Object {
            "fontWeight": "bold",
          }
        }
        theme={
          Object {
            "colors": Object {
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "primary": "#007aff",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          }
        }
      >
        Currency
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "alignSelf": "center",
          "borderBottomWidth": 0.5,
          "flexWrap": "nowrap",
          "height": 25,
          "justifyContent": "center",
          "overflow": "hidden",
          "paddingBottom": 3,
          "paddingLeft": 5,
          "paddingRight": 5,
          "width": "25%",
        }
      }
    >
      <Text
        style={
          Object {
            "fontWeight": "bold",
          }
        }
        theme={
          Object {
            "colors": Object {
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "primary": "#007aff",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          }
        }
      >
        Balance
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "alignSelf": "center",
          "borderBottomWidth": 0.5,
          "flexWrap": "nowrap",
          "height": 25,
          "justifyContent": "center",
          "overflow": "hidden",
          "paddingBottom": 3,
          "paddingLeft": 5,
          "paddingRight": 5,
          "width": "25%",
        }
      }
    >
      <Text
        style={
          Object {
            "fontWeight": "bold",
          }
        }
        theme={
          Object {
            "colors": Object {
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "primary": "#007aff",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          }
        }
      >
        AUD Price
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "alignSelf": "center",
          "borderBottomWidth": 0.5,
          "flexWrap": "nowrap",
          "height": 25,
          "justifyContent": "center",
          "overflow": "hidden",
          "paddingBottom": 3,
          "paddingLeft": 5,
          "paddingRight": 5,
          "width": "25%",
        }
      }
    >
      <Text
        style={
          Object {
            "fontWeight": "bold",
          }
        }
        theme={
          Object {
            "colors": Object {
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "primary": "#007aff",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          }
        }
      >
        AUD Value
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-evenly",
        "paddingTop": 2,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flexWrap": "nowrap",
          "height": 15,
          "justifyContent": "center",
          "overflow": "hidden",
          "paddingLeft": 5,
          "paddingRight": 5,
          "width": "25%",
        }
      }
    >
      <Text
        style={
          Object {
            "flexWrap": "nowrap",
            "fontFamily": "Courier New",
            "fontSize": 11,
            "overflow": "hidden",
          }
        }
        theme={
          Object {
            "colors": Object {
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "primary": "#007aff",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          }
        }
      >
        Xbt
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flexWrap": "nowrap",
          "height": 15,
          "justifyContent": "center",
          "overflow": "hidden",
          "paddingLeft": 5,
          "paddingRight": 5,
          "width": "25%",
        }
      }
    >
      <Text
        style={
          Object {
            "flexWrap": "nowrap",
            "fontFamily": "Courier New",
            "fontSize": 11,
            "overflow": "hidden",
          }
        }
        theme={
          Object {
            "colors": Object {
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "primary": "#007aff",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          }
        }
      >
         3         
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flexWrap": "nowrap",
          "height": 15,
          "justifyContent": "center",
          "overflow": "hidden",
          "paddingLeft": 5,
          "paddingRight": 5,
          "width": "25%",
        }
      }
    >
      <Text
        style={
          Object {
            "flexWrap": "nowrap",
            "fontFamily": "Courier New",
            "fontSize": 11,
            "overflow": "hidden",
          }
        }
        theme={
          Object {
            "colors": Object {
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "primary": "#007aff",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          }
        }
      >
        16,839.25
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flexWrap": "nowrap",
          "height": 15,
          "justifyContent": "center",
          "overflow": "hidden",
          "paddingLeft": 5,
          "paddingRight": 5,
          "width": "25%",
        }
      }
    >
      <Text
        style={
          Object {
            "flexWrap": "nowrap",
            "fontFamily": "Courier New",
            "fontSize": 11,
            "overflow": "hidden",
          }
        }
        theme={
          Object {
            "colors": Object {
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "primary": "#007aff",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          }
        }
      >
        50,517.75
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-evenly",
        "paddingTop": 2,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flexWrap": "nowrap",
          "height": 15,
          "justifyContent": "center",
          "overflow": "hidden",
          "paddingLeft": 5,
          "paddingRight": 5,
          "width": "25%",
        }
      }
    >
      <Text
        style={
          Object {
            "flexWrap": "nowrap",
            "fontFamily": "Courier New",
            "fontSize": 11,
            "overflow": "hidden",
          }
        }
        theme={
          Object {
            "colors": Object {
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "primary": "#007aff",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          }
        }
      >
        Eth
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flexWrap": "nowrap",
          "height": 15,
          "justifyContent": "center",
          "overflow": "hidden",
          "paddingLeft": 5,
          "paddingRight": 5,
          "width": "25%",
        }
      }
    >
      <Text
        style={
          Object {
            "flexWrap": "nowrap",
            "fontFamily": "Courier New",
            "fontSize": 11,
            "overflow": "hidden",
          }
        }
        theme={
          Object {
            "colors": Object {
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "primary": "#007aff",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          }
        }
      >
        21.62806365
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flexWrap": "nowrap",
          "height": 15,
          "justifyContent": "center",
          "overflow": "hidden",
          "paddingLeft": 5,
          "paddingRight": 5,
          "width": "25%",
        }
      }
    >
      <Text
        style={
          Object {
            "flexWrap": "nowrap",
            "fontFamily": "Courier New",
            "fontSize": 11,
            "overflow": "hidden",
          }
        }
        theme={
          Object {
            "colors": Object {
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "primary": "#007aff",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          }
        }
      >
           464.04
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flexWrap": "nowrap",
          "height": 15,
          "justifyContent": "center",
          "overflow": "hidden",
          "paddingLeft": 5,
          "paddingRight": 5,
          "width": "25%",
        }
      }
    >
      <Text
        style={
          Object {
            "flexWrap": "nowrap",
            "fontFamily": "Courier New",
            "fontSize": 11,
            "overflow": "hidden",
          }
        }
        theme={
          Object {
            "colors": Object {
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "primary": "#007aff",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          }
        }
      >
        10,036.29
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-evenly",
        "paddingTop": 2,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "alignSelf": "center",
          "borderBottomWidth": 0.5,
          "borderTopWidth": 0.5,
          "flexWrap": "nowrap",
          "height": 20,
          "justifyContent": "center",
          "overflow": "hidden",
          "paddingBottom": 5,
          "paddingLeft": 5,
          "paddingRight": 5,
          "paddingTop": 3,
          "width": "25%",
        }
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "Courier New",
            "fontSize": 11,
            "fontWeight": "bold",
          }
        }
        theme={
          Object {
            "colors": Object {
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "primary": "#007aff",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          }
        }
      >
        Total
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "alignSelf": "center",
          "borderBottomWidth": 0.5,
          "borderTopWidth": 0.5,
          "flexWrap": "nowrap",
          "height": 20,
          "justifyContent": "center",
          "overflow": "hidden",
          "paddingBottom": 5,
          "paddingLeft": 5,
          "paddingRight": 5,
          "paddingTop": 3,
          "width": "25%",
        }
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "Courier New",
            "fontSize": 11,
            "fontWeight": "bold",
          }
        }
        theme={
          Object {
            "colors": Object {
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "primary": "#007aff",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          }
        }
      >
        
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "alignSelf": "center",
          "borderBottomWidth": 0.5,
          "borderTopWidth": 0.5,
          "flexWrap": "nowrap",
          "height": 20,
          "justifyContent": "center",
          "overflow": "hidden",
          "paddingBottom": 5,
          "paddingLeft": 5,
          "paddingRight": 5,
          "paddingTop": 3,
          "width": "25%",
        }
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "Courier New",
            "fontSize": 11,
            "fontWeight": "bold",
          }
        }
        theme={
          Object {
            "colors": Object {
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "primary": "#007aff",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          }
        }
      >
        
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "alignSelf": "center",
          "borderBottomWidth": 0.5,
          "borderTopWidth": 0.5,
          "flexWrap": "nowrap",
          "height": 20,
          "justifyContent": "center",
          "overflow": "hidden",
          "paddingBottom": 5,
          "paddingLeft": 5,
          "paddingRight": 5,
          "paddingTop": 3,
          "width": "25%",
        }
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "Courier New",
            "fontSize": 11,
            "fontWeight": "bold",
          }
        }
        theme={
          Object {
            "colors": Object {
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "primary": "#007aff",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          }
        }
      >
        60,554.04
      </Text>
    </View>
  </View>
</View>
`;
